### PR TITLE
Adding alternate key support for app entity

### PR DESCRIPTION
**Pull request by aricrowe57**

### CODE REVIEW

**Comment by aricrowe57** [comment]
Hi @FaithOmbongi , we are adding 'alternate key support' for the application entity, so the appId can be used as an alternate key (instead of just object ID).  Hopefully this docs PR is explanatory enough of what that is.

This PR is still in draft state, but I'm hoping to get a few questions answered before progressing any further on it. 

1. Do you think the example HTTP requests, where I show making the same HTTP call with the primary key syntax and alternate key syntax, should be replicated across all application methods (i.e., should I also add similar info to 'addKey' and 'removeKey', etc.)
2. If so, are you able to review what I have so I don't replicate some bad updates across various different articles?
3. Any other comments on what we have so far?

Thanks!

File: api-reference/beta/api/application-delete.md [hunk 1]
```diff
@@ -25,12 +25,22 @@ One of the following permissions is required to call this API. To learn more, in
...
 <!-- { "blockType": "ignored" } -->
 ```http
 DELETE /applications/{id}
 ```
 
+### Application ID Key
+Replace `{appId}` with the **appId** for the application object, also referred to as the **Application (Client) ID** in the Azure portal.
+<!-- { "blockType": "ignored" } -->
+```http
+DELETE /applications/(appId='{appId}')
```
Comment (FaithOmbongi) [inline api-reference/beta/api/application-delete.md:41]: Hi @aricrowe57 - A globally applicable suggestion to make the block shorter.
```suggestion
You can address the application using either its **id** or **appId**. **id** and **appId** are referred to as the **Object ID** and **Application (Client) ID** respectively, in the Azure portal.
<!-- { "blockType": "ignored" } -->
```http
DELETE /applications/{id}
DELETE /applications/(appId='{appId}')
```

**Comment by FaithOmbongi** [comment]
> Hi @FaithOmbongi , we are adding 'alternate key support' for the application entity, so the appId can be used as an alternate key (instead of just object ID). Hopefully this docs PR is explanatory enough of what that is.
> 
> This PR is still in draft state, but I'm hoping to get a few questions answered before progressing any further on it.
> 
> 1. Do you think the example HTTP requests, where I show making the same HTTP call with the primary key syntax and alternate key syntax, should be replicated across all application methods (i.e., should I also add similar info to 'addKey' and 'removeKey', etc.)
> 2. If so, are you able to review what I have so I don't replicate some bad updates across various different articles?
> 3. Any other comments on what we have so far?
> 
> Thanks!

Hi @aricrowe57

1.) Yes - the HTTP request syntax should be replicated in all docs that support it. 
           Q: Is this for both beta & v1.0 or is the capability for beta only?

2,3.) I also recommend adding a changelog so this capability doesn't pass under the radar. A suggested changelog object 👇🏾. If this capability applies to v1.0 as well, please add another object referencing v1.0 (use a different ID GUID).

```json
{
    "ChangeList": [
      {
        "Id": "b1b69af7-630f-4183-a152-ebffa1a4bf27",
        "ApiChange": "Property",
        "ChangedApiName": "appId",
        "ChangeType": "Addition",
        "Description": "Added support for addressing an [application](https://docs.microsoft.com/graph/api/resources/application?view=graph-rest-beta) using the application **appId** as the alternate key. You can now address an application via either the primary key **id** or the alternate key **appId** as the path parameter.",
        "Target": "application"
      }
    ],
    "Id": "b1b69af7-630f-4183-a152-ebffa1a4bf27",
    "Cloud": "Prod",
    "Version": "beta",
    "CreatedDateTime": "2021-08-19T00:00:00.077762Z",
    "WorkloadArea": "Application",
    "SubArea": ""
}
```

**Comment by aricrowe57** [comment]
@FaithOmbongi , to answer your question above - this is just for /beta for now.  Will do another PR for /v1.0 soon.  This is ready for review.  Thanks!

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Hi @aricrowe57 - Just minor fixes/oversights to fix.

I notice this capability hasn't yet been deployed to beta so we can't test. The schema PR also related to PPE, not Prod-beta. Please only merge the PR when the capability and schema are live.

**Comment by FaithOmbongi** [comment]
Hi @aricrowe57 - This PR has review feedback yet to be addressed, all minor fixes. Please look at addressing them. Meanwhile, your PR is attracting merge conflicts in the changelog.

**Comment by FaithOmbongi** [comment]
Hi @aricrowe57 - I realize I'd requested the wrong changes in my previous review, confusing application docs with serviceprincipal docs. My bad 🤦🏾‍♀️. Only one comment remains valid and your PR will be good to go once the capability is live.

PR https://github.com/microsoftgraph/microsoft-graph-docs/pull/18260#pullrequestreview-1100873635 introduces changes to service principals.

**Comment by FaithOmbongi** [comment]
Heads up @aricrowe57 - I see the schema changes are now live.

```
 <Annotation Term="OData.Community.Keys.V1.AlternateKeys">
          <Collection>
            <Record Type="OData.Community.Keys.V1.AlternateKey">
              <PropertyValue Property="Key">
                <Collection>
                  <Record Type="OData.Community.Keys.V1.PropertyRef">
                    <PropertyValue Property="Alias" String="appId" />
                    <PropertyValue Property="Name" PropertyPath="appId" />
                  </Record>
                </Collection>
              </PropertyValue>
            </Record>
          </Collection>
        </Annotation>
```

Please get this PR ready to merge. Thank you.

**Review by FaithOmbongi — APPROVED** [review]
Pushed commits to:

- Fix syntax
- Update changelog date

**Comment by aricrowe57** [comment]
#sign-off

**Pull request merged**